libraries/GCS_MAVLink/GCS.h

```diff
@@ -391,3 +365,58 @@ class GCS_MAVLINK
     bool signing_enabled(void) const;
     static void save_signing_timestamp(bool force_save_now);
 };
+
+/// @class GCS
+/// @brief global GCS object
+class GCS
+{
+
+public:
+
+    GCS() {
+        if (_singleton  == nullptr) {
+            _singleton = this;
+        } else {
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+            // this is a serious problem, but we don't need to kill a
+            // real vehicle
+            AP_HAL::panic("GCS must be singleton");
+#endif
+        }
+    };
+
+    static class GCS *instance() {
+        return _singleton;
```

Comment:
if instance() is called before the object is constructed, this will lead to  crash. You should construct the object in this method and make the constructor private. It's actually very easy to trigger this problem when you are initializing things in the constructors of static objects.